### PR TITLE
Resolving cluster dup

### PR DESCRIPTION
**Pull request by jaysim**
1. check cluster duplication corruption.
2. resolve cluster duplication corruption with allocating new cluster and copying duplicated clusters data to new.
3. optimization checking logic.
4. fix some bugs.

### CODE REVIEW

File: src/utils.h [hunk 1]
```diff
@@ -96,7 +96,8 @@ int mft_next_record(struct mft_search_ctx *ctx);
 
 /* MAX_PATH definition was missing in ntfs-3g's headers. */
 #ifndef MAX_PATH
-#define MAX_PATH 1024
+#define MAX_PATH 8192
+//#define MAX_PATH 1024
```
Comment (namjaejeon) [inline src/utils.h]: Why did you not remove MAX_PATH here ?

**Comment by namjaejeon** [comment]
Can you run checkpatch.pl against this PR ?

```
WARNING: Statements terminations use 1 semicolon
#47: FILE: src/ntfsck.c:4026:
+	ntfs_inode *ni = NULL;;

ERROR: code indent should use tabs where possible
#47: FILE: libntfs-3g/runlist.c:926:
+                               }$

WARNING: please, no spaces at the start of a line
#47: FILE: libntfs-3g/runlist.c:926:
+                               }$

ERROR: code indent should use tabs where possible
#48: FILE: libntfs-3g/runlist.c:927:
+                        }$

WARNING: please, no spaces at the start of a line
#48: FILE: libntfs-3g/runlist.c:927:
+                        }$
```

File: src/utils.h (hunk 1 above)
Comment (jaysim) [reply src/utils.h]: it's temporary code for test. I'll revert this file.

**Comment by namjaejeon** [comment]
Applied, Thanks for your work!

**Pull request closed without merging**